File: utils/utils-test/test/com/xenoage/utils/iterators/ReverseIteratorTest.java
package com.xenoage.utils.iterators;

import org.junit.Test;

import java.util.ArrayList;

import static org.junit.Assert.*;

/**
 * Tests for {@link ReverseIterator}.
 *
 * @author Andreas Wenger
 */
public class ReverseIteratorTest {

	@Test public void test() {
		int count = 5;
		ArrayList<Integer> a = new ArrayList<>(count);
		for (int i = 0; i < count; i++)
			a.add(i);
		for (int i : new ReverseIterator<>(a))
			assertEquals(--count, i);
	}

}

File: utils-kotlin/utils-kotlin-jvm/test/com/xenoage/utils/iterators/ReverseIteratorTest.kt
package com.xenoage.utils.iterators

import kotlin.test.Test
import kotlin.test.assertEquals

/**
 * Tests for [ReverseIterator].
 */
class ReverseIteratorTest {

	@Test fun test() {
		var count = 5
		val a = (0..count).toList()
		for (i in a.reverseIt())
			assertEquals(count--, i)
	}

}
